src/forder.c

```diff
@@ -1526,7 +1526,11 @@ SEXP forderLazy(SEXP DT, SEXP by, SEXP retGrpArg, SEXP sortGroupsArg, SEXP ascAr
           (hasGrp && retGrp)) {
         opt = 2; // idxOpt retGrp==hasGrp, if condition unfolded for codecov
       } else if (hasGrp && !retGrp) {
-        idx = copyAsPlain(idx);
+        // shallow_duplicate is faster than copyAsPlain, but shallow_duplicate is AFAIK good for VECSXP, not for INTSXP
+        // it is still the bottleneck in this opt, it is now better to call retGrp=TRUE and just not use those extra arguments
+        // can we do better here? real shallow for INTSXP? If we could just re-point data pointer... like we do for DT columns
+        // SEXP new; INTEGER(new) = INTEGER(idx); setAttrib(new, ..., R_NilValue)
+        idx = shallow_duplicate(idx);
```

Comment:
using data from #4380 
```
#before this change
forderLazy: opt=2, took 0.515s
# after this change
forderLazy: opt=2, took 0.258s
```
still, it doesn't seems reasonable to waste 0.25s just for re-using existing index, any ideas? @mattdowle

Reply:
it is made like this to strip out `starts` and `maxgrpn` attributes when calling `retGrp=F`.

Reply:
https://github.com/Rdatatable/data.table/issues/4467 this is request to address that